app/components/member_list/index.js

```diff
@@ -67,16 +72,29 @@ export default class MemberList extends PureComponent {
             rowHasChanged: (r1, r2) => r1 !== r2,
             sectionHeaderHasChanged: (s1, s2) => s1 !== s2
         });
-        const dataSource = props.sections ? ds.cloneWithRowsAndSections(this.createSections(props.members)) : ds.cloneWithRows(props.members);
+        let data = props.members;
+        if (props.showSections) {
+            data = this.createSections(props.members);
+        }
+        const dataSource = ds.cloneWithRowsAndSections(data);
         this.state = {
+            data,
             dataSource
         };
     }
 
     componentWillReceiveProps(nextProps) {
-        const {members, sections} = nextProps;
-        const dataSource = sections ? this.state.dataSource.cloneWithRowsAndSections(this.createSections(members)) : this.state.dataSource.cloneWithRows(members);
+        const {members, showSections} = nextProps;
+
+        let data = members;
```

Comment:
Is it possible to only update the data when `members` or `showSections` changes? This looks like it'll trigger and regenerate the data any time the redux store updates